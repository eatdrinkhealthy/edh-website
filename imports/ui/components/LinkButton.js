import React, {
  PropTypes,
} from "react";
import { linkButtonStyle } from "../styles";

const LinkButton = props => (
  <a href={props.url} className={linkButtonStyle.className}>{props.text}</a>
);

LinkButton.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default LinkButton;
