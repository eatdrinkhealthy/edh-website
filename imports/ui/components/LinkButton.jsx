import React from "react";
import PropTypes from "prop-types";
import { linkButtonStyle } from "../styles";

const LinkButton = props => (
  <a href={props.url} className={linkButtonStyle.className}>
    {props.text}
  </a>
);

LinkButton.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default LinkButton;
