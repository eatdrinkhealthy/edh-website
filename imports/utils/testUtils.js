/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import cheerio from "cheerio";
import _ from "lodash";
import {
  mount,
} from "enzyme";

export function getComponentClasses(currComponent) {
  // NOTE: using mount here to ensure all child elements are fully rendered
  //       but is seems shallow might work too
  const mountedHTML = mount(currComponent).html();
  console.log("html:", mountedHTML);

  const $ = cheerio.load(mountedHTML);
  let classList = [];
  $("*").each(function () {
    // add new unique classes of any element
    classList = _.union(classList, $(this).attr("class").split(" "));
  });
  return classList;
}
