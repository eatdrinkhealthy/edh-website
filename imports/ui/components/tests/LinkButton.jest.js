/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import LinkButton from "../LinkButton";
import { formattedJSON } from "../../../utils/tests/testUtils";

describe("<LinkButton />", function () {
  it("render matches snapshot", function () {
    const tree = renderer.create(<LinkButton text="some button text" url="some link" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("css matches snapshot", function () {
    expect(shallow(<LinkButton text="" url="" />).hasClass(LinkButton.style.className)).toBe(true);
    expect(formattedJSON(LinkButton.style)).toMatchSnapshot();
  });
});
