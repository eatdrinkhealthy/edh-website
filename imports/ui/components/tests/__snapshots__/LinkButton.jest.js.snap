exports[`<LinkButton /> matches render snapshot 1`] = `
<a
  className="css-15br13v"
  href="some link">
  some button text
</a>
`;

exports[`<LinkButton /> matches style / css snapshot 1`] = `
"[
  {
    \"rules\": {
      \"backgroundColor\": \"darkslategray\",
      \"border\": \"none\",
      \"borderRadius\": \"4px\",
      \"color\": \"white\",
      \"padding\": \"8px 32px 8px 32px\",
      \"textAlign\": \"center\",
      \"textDecoration\": \"none\",
      \"fontSize\": \"12px\"
    },
    \"className\": \"css-15br13v\"
  }
]"
`;
